src/app/store/trips/effects.ts

```diff
@@ -83,6 +87,8 @@ export class TripsEffects {
               duration: 5000,
               panelClass: ['snackbar-success'],
             });
+            this.router.navigate(['/trip', trip.docId]);
+            this.store.dispatch(setSelectedTripId({ tripId: trip.docId }));
```

Comment:
This is a gross way to do this IMO. 
The other thing you can do is instead of a switchmap, I think you return a concatmap and then you return an array of actions to dispatch

Reply:
Thank you. Will defs change it